lib/licensed/sources/gradle.rb

```diff
@@ -4,132 +4,164 @@
-      # Defaults to ["runtime", "runtimeClasspath"]
-      def configurations
-        @configurations ||= begin
-          if configurations = config.dig("gradle", "configurations")
-            Array(configurations)
-          else
-            DEFAULT_CONFIGURATIONS
+      class Dependency < Licensed::Dependency
+        GRADLE_LICENSES_CSV_NAME = "licenses.csv".freeze
+
+        class << self
+          # Returns a key to uniquely identify a name and version in the obtained CSV content
+          def csv_key(name:, version:)
+            "#{name}-#{version}"
+          end
+
+          # Loads and caches license report CSV data as a hash of :name-:version => :url pairs
+          #
+          # executable     - The gradle executable to run to generate the license report
+          # configurations - The gradle configurations to generate license report for
+          #
+          # Returns a hash of dependency identifiers to their license content URL
+          def load_csv(path, executable, configurations)
+            @csv ||= begin
+              gradle_licenses_dir = File.join(path, GRADLE_LICENSES_PATH)
+              Licensed::Sources::Gradle.gradle_command("generateLicenseReport", path: path, executable: executable, configurations: configurations)
+              CSV.foreach(File.join(gradle_licenses_dir, GRADLE_LICENSES_CSV_NAME), headers: true).each_with_object({}) do |row, hsh|
+                name, _, version = row["artifact"].rpartition(":")
+                key = csv_key(name: name, version: version)
+                hsh[key] = row["moduleLicenseUrl"]
+              end
+            ensure
+              FileUtils.rm_rf(gradle_licenses_dir)
+            end
           end
+
+          # Returns the cached url for the given dependency
+          def url_for(dependency)
+            @csv[csv_key(name: dependency.name, version: dependency.version)]
+          end
+
+          # Cache and return the results of getting the license content.
+          def retrieve_license(url)
+            (@licenses ||= {})[url] ||= Net::HTTP.get(URI(url))
+          end
+        end
+
+        def initialize(name:, version:, path:, executable:, configurations:, metadata: {})
+          @configurations = configurations
+          @executable = executable
+          super(name: name, version: version, path: path, metadata: metadata)
+        end
+
+        # Returns whether the dependency content exists
+        def exist?
+          # shouldn't force network connections just to check if content exists
+          # only check that the path is not empty
+          !path.to_s.empty?
+        end
+
+        # Returns a Licensee::ProjectFiles::LicenseFile for the dependency
+        def project_files
+          self.class.load_csv(path, @executable, @configurations)
+
+          url = self.class.url_for(self)
+          license_data = self.class.retrieve_license(url)
+          Array(Licensee::ProjectFiles::LicenseFile.new(license_data, { uri: url }))
```

Comment:
with the new `uri` property that's picked up in `Licensed::Dependency` there's no need to write a file to disk with the license data 👍